cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinPatchEventConsumerTest.java

```diff
@@ -31,39 +54,157 @@ public class CristinPatchEventConsumerTest extends AbstractCristinImportTest {
         outputStream = new ByteArrayOutputStream();
     }
 
     @Test
-    void dummyTest() throws IOException {
+    void shouldThrowExceptionWhenChildPublicationCannotBeRetrieved() throws ApiGatewayException, IOException {
+        var partOfCristinId = randomString();
+        var childPublicationIdentifier = SortableIdentifier.next();
+        createPersistedPublicationWithStatusPublishedWithSpecifiedCristinId(partOfCristinId);
+        var partOfEventReference = createPartOfEventReference(childPublicationIdentifier.toString(), partOfCristinId);
+        var fileUri = s3Driver.insertFile(randomPath(), partOfEventReference);
+        var eventReference = createInputEventForFile(fileUri);
+        var input = toInputStream(eventReference);
+        Executable action = () -> handler.handleRequest(input, outputStream, CONTEXT);
+        assertThrows(NotFoundException.class, action);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenSearchingForNvaPublicationByCristinIdentifierReturnsMoreThanOnePublication()
+        throws ApiGatewayException, IOException {
+        var childPublication = createPersistedPublicationWithStatusPublishedWithSpecifiedCristinId(randomString());
+        var partOfCristinId = randomString();
+        persistSeveralPublicationsWithTheSameCristinId(partOfCristinId);
+        var partOfEventReference = createPartOfEventReference(childPublication.getIdentifier().toString(),
+                                                              partOfCristinId);
+        var fileUri = s3Driver.insertFile(randomPath(), partOfEventReference);
+        var eventReference = createInputEventForFile(fileUri);
+        var input = toInputStream(eventReference);
+        Executable action = () -> handler.handleRequest(input, outputStream, CONTEXT);
+        var exception = assertThrows(ParentPublicationException.class, action);
+        assertThat(exception.getMessage(), containsString(INVALID_PARENT_MESSAGE));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenSearchingForNvaPublicationByCristinIdentifierReturnsNoPublication()
+        throws ApiGatewayException, IOException {
+        var childPublication = createPersistedPublicationWithStatusPublishedWithSpecifiedCristinId(randomString());
+        var partOfCristinId = randomString();
+        var partOfEventReference = createPartOfEventReference(childPublication.getIdentifier().toString(),
+                                                              partOfCristinId);
+        var fileUri = s3Driver.insertFile(randomPath(), partOfEventReference);
+        var eventReference = createInputEventForFile(fileUri);
+        var input = toInputStream(eventReference);
+        Executable action = () -> handler.handleRequest(input, outputStream, CONTEXT);
+        var exception = assertThrows(ParentPublicationException.class, action);
+        assertThat(exception.getMessage(), containsString(INVALID_PARENT_MESSAGE));
+    }
```

Comment:
We always expect that there exists publication when cristinIdentifier is in partOf?